### PR TITLE
OverconstrainedError | remove {{securecontext_header}} & update inherits information

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

reference to https://w3c.github.io/mediacapture-main#dom-overconstrainederror

remove {{securecontext_header}}, the interface is also available in insecure context

fix `OverconstrainedError()`, mark `message` with {{optional_inline}} and add default value description

add inherits information of `OverconstrainedError`, also remove inherits properties and methods

some style update

### CODE REVIEW

File: files/en-us/web/api/overconstrainederror/overconstrainederror/index.md [hunk 1]
```diff
@@ -25,8 +25,8 @@ new OverconstrainedError()
 
 - `constraint`
   - : The constraint that was not satisfied.
-- `message`
-  - : Text for the error's `message` property.
+- `message` {{optional_inline}}
+  - : Text for the error's `message` property. It defaults to be an empty string.
```
Comment (wbamberg) [inline files/en-us/web/api/overconstrainederror/overconstrainederror/index.md:29]:
```suggestion
  - : Text for the error's `message` property. ] Defaults to an empty string.
```

File: files/en-us/web/api/overconstrainederror/index.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ page-type: web-api-interface
 browser-compat: api.OverconstrainedError
 ---
 
-{{securecontext_header}}{{APIRef("Media Capture and Streams")}}
+{{APIRef("Media Capture and Streams")}}
 
-The **`OverconstrainedError`** interface of the [Media Capture and Streams API](/en-US/docs/Web/API/Media_Capture_and_Streams_API) indicates that the set of desired capabilities for the current {{domxref('MediaStreamTrack')}} cannot currently be met. When this event is thrown on a MediaStreamTrack, it is muted until either the current constraints can be established or until satisfiable constraints are applied.
+The **`OverconstrainedError`** interface of the {{domxref("Media Capture and Streams API", "", "", "nocode")}} indicates that the set of desired capabilities for the current {{domxref('MediaStreamTrack')}} cannot currently be met. When this event is thrown on a MediaStreamTrack, it is muted until either the current constraints can be established or until satisfiable constraints are applied.
```
Comment (wbamberg) [inline files/en-us/web/api/overconstrainederror/index.md:10]: Different people do different things here, but personally I think it is better just to use a Markdown link when linking to an API overview page, as it's not a concrete part of an interface, and forces us into this `"", "", "nocode"` thing.

Comment (skyclouds2001) [inline files/en-us/web/api/overconstrainederror/index.md:10]:
```suggestion
The **`OverconstrainedError`** interface of the [Media Capture and Streams API](/en-US/docs/Web/API/Media_Capture_and_Streams_API) indicates that the set of desired capabilities for the current {{domxref('MediaStreamTrack')}} cannot currently be met. When this event is thrown on a MediaStreamTrack, it is muted until either the current constraints can be established or until satisfiable constraints are applied.
```

File: files/en-us/web/api/overconstrainederror/overconstrainederror/index.md (hunk 1 above)
Comment (skyclouds2001) [inline files/en-us/web/api/overconstrainederror/overconstrainederror/index.md:29]:
```suggestion
  - : Text for the error's `message` property. Defaults to an empty string.
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @skyclouds2001 !

**Pull request merged**